include/z64math.h

```diff
@@ -112,13 +112,28 @@ typedef struct {
 
 #define IS_ZERO(f) (fabsf(f) < 0.008f)
 
+// Casting a float to an integer, when the float value is larger than what the integer type can hold,
+// leads to undefined behavior. For example (f32)0x8000 doesn't fit in a s16, so it cannot be cast to s16.
+// This isn't an issue with IDO, but is one with for example GCC.
+// A partial workaround is to cast to s32 then s16, hoping all binang values used will fit a s32.
+#define TRUNCF_BINANG(f) (s16)(s32)(f)
+
 // Angle conversion macros
-#define DEG_TO_BINANG(degrees) (s16)((degrees) * (0x8000 / 180.0f))
-#define RADF_TO_BINANG(radf) (s16)((radf) * (0x8000 / M_PI))
-#define RADF_TO_DEGF(radf) ((radf) * (180.0f / M_PI))
-#define DEGF_TO_RADF(degf) ((degf) * (M_PI / 180.0f))
-#define BINANG_TO_RAD(binang) ((f32)binang * (M_PI / 0x8000))
-#define BINANG_TO_RAD_ALT(binang) (((f32)binang / 0x8000) * M_PI)
+#define DEG_TO_RAD(degrees) ((degrees) * (M_PI / 180.0f))
+#define DEG_TO_BINANG(degrees) TRUNCF_BINANG((degrees) * (0x8000 / 180.0f))
+#define DEG_TO_BINANG_ALT(degrees) TRUNCF_BINANG(((degrees) / 180.0f) * 0x8000)
+#define DEG_TO_BINANG_ALT2(degrees) TRUNCF_BINANG(((degrees) * 0x10000) / 360.0f)
+#define DEG_TO_BINANG_ALT3(degrees) ((degrees) * (0x8000 / 180.0f))
```

Comment:
So `DEG_TO_BINANG_ALT3` exists because `TRUNCF_BINANG` breaks matching for `DEG_TO_BINANG`?

Reply:
Yes, but more specifically it is the `(s16)` cast. The ones that use it cast things differently (most just use `s32` though there is one instance of no cast i.e. keep as f32).

Worth mentioning, (and probably should have mentioned up above my bad), but it is possible to change the macros to just cast to `s32`s with just a few instances of needing to explicitly cast to an `s16` (3 files). Afaik all instances are used in conjunction with an `s16` so will implicitly get cast at some point. Than `DEG_TO_BINANG_ALT3` would only have the one use, so would probably just be removed and a comment left there. I'm not sure this is a great idea since binangs should always be cast to an `s16` for overflow to work properly, but they should also be used with `s16` variables so maybe it fine 🤷.